src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -569,6 +569,7 @@ private BoundExpression CreateFunctionTypeConversion(SyntaxNode syntax, BoundExp
             }
 
             conversion = Conversions.ClassifyConversionFromExpression(source, delegateType, ref useSiteInfo);
+            bool warnOnMethodGroupConversion = source.Kind == BoundKind.MethodGroup && conversion.Exists && destination.SpecialType == SpecialType.System_Object;
```

Comment:
This doesn't check whether the conversion is implicit, is that intended?

Reply:
I would expect us to be checking `isCast`. If we're not, there's probably some test case that's missing.

Reply:
> This doesn't check whether the conversion is implicit

Good catch, thanks.